launch/tests/next_schedule_test.rs

```diff
@@ -117,52 +184,338 @@ fn test_no_pickup_dropoff(fixture_model: BaseModel) -> Result<(), Error> {
+
+    // Departure Tests
+    {
+        let mut config = ScheduleConfig::new(
+            ScheduleOn::BoardTimes,
+            "stop_area:sa:J",
+            "2020-01-02T00:00:00",
+        );
+        config.duration = 10 * 60; // 10m
+        let result = build_and_solve_schedule(&config, &fixture_model)?;
+        assert_eq!(result.len(), 1);
+        let stop_time = &result[0];
+        assert_eq!(stop_time.time, "2020-01-02T00:10:00".as_datetime());
+
+        config.duration = 30 * 60; // 30m
+        let result = build_and_solve_schedule(&config, &fixture_model)?;
+        assert_eq!(result.len(), 2);
+        let stop_time = &result[1];
+        assert_eq!(stop_time.time, "2020-01-02T00:30:00".as_datetime());
+    }
+
+    // Arrival Tests
+    {
+        let mut config = ScheduleConfig::new(
+            ScheduleOn::DebarkTimes,
+            "stop_area:sa:J",
+            "2020-01-02T00:00:00",
+        );
+        config.duration = 10 * 60; // 10m
+        let result = build_and_solve_schedule(&config, &fixture_model)?;
+        assert_eq!(result.len(), 1);
+        let stop_time = &result[0];
+        assert_eq!(stop_time.time, "2020-01-02T00:10:00".as_datetime());
+
+        config.duration = 30 * 60; // 30m
+        let result = build_and_solve_schedule(&config, &fixture_model)?;
+        assert_eq!(result.len(), 2);
+        let stop_time = &result[1];
+        assert_eq!(stop_time.time, "2020-01-02T00:30:00".as_datetime());
+    }
+
+    Ok(())
+}
+
+#[rstest]
+fn test_on_route(fixture_model: BaseModel) -> Result<(), Error> {
+    let _log_guard = launch::logger::init_test_logger();
+
+    // In this test we expect to receive all departure of all stop_points of route R1
+    // after a certain datetime (we test multiple datetime)
+
+    // Departure Tests
+    {
+        let mut config =
+            ScheduleConfig::new(ScheduleOn::BoardTimes, "route:R1", "2020-01-01T10:00:00");
+        config.duration = 10 * 60 * 60; // 10h
+
+        // on route R1 we have 9 stop_points and 6 valid vj per day
+        // we expect 6 Next Departure per day with from_datetime = 2020-01-02T10:00:00
+        // At stop_point A & B
+        // Note: Last stop_point of a VJ is a terminus and considered as not bordable
+        let result = build_and_solve_schedule(&config, &fixture_model)?;
+        assert_eq!(result.len(), 4);
+
+        // We update from_datetime to retrieve departures after 2020-01-01T11:00:00
+        config.from_datetime = "2020-01-01T11:00:00".as_datetime();
+        let result = build_and_solve_schedule(&config, &fixture_model)?;
+        assert_eq!(result.len(), 2);
+    }
+
+    // Arrival Tests
+    {
+        let mut config =
+            ScheduleConfig::new(ScheduleOn::DebarkTimes, "route:R1", "2020-01-01T10:00:00");
+        config.duration = 10 * 60 * 60; // 10h
+
+        // on route R1 we have 9 stop_points and 6 valid vj per day
+        // we expect 6 Next Departure per day with from_datetime = 2020-01-02T10:00:00
```

Comment:
```suggestion
        // on route R1 we have 3 stop_points and 2 valid vj per day
        // we expect 4 Next Departure per day with from_datetime = 2020-01-02T10:00:00
```
as above, why 4 is the expected result ?